azext_iot/tests/central/__init__.py

```diff
@@ -148,49 +185,52 @@ def _create_device_template(self):
         command = "iot central device-template create --app-id {} --device-template-id {} -k '{}'".format(
             APP_ID, template_id, device_template_path
         )
-        command = self._appendOptionalArgsToCommand(command, TOKEN, DNS_SUFFIX)
 
-        result = self.cmd(command, checks=[self.check("displayName", template_name), ],)
+        result = self.cmd(
+            command,
+            api_version=api_version,
+            checks=[
+                self.check("displayName", template_name),
+            ],
+        )
         json_result = result.get_output_in_json()
 
         assert json_result["@id"] == template_id
         return (template_id, template_name)
 
-    def _delete_device_template(self, template_id):
+    def _delete_device_template(self, api_version, template_id):
         attempts = range(0, 10)
         command = "iot central device-template delete --app-id {} --device-template-id {}".format(
             APP_ID, template_id
         )
 
-        command = self._appendOptionalArgsToCommand(command, TOKEN, DNS_SUFFIX)
-
         # retry logic to delete the template
         for _ in attempts:
             try:
-                self.cmd(command, checks=[self.check("result", "success")])
+                self.cmd(
+                    command,
+                    api_version=api_version,
+                    checks=[self.check("result", "success")],
+                )
                 return
             except:
                 time.sleep(10)
```

Comment:
with the current logic if retry fails after n attempts user does not get indication about failure

Reply:
fixed